### ISSUE TITLE
VPC network tier DNS doesn't reflect VPC DNS settings

### ISSUE THREAD

**Issue by alexandremattioli** [apache/cloudstack#8598]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
VPC, VR
~~~

##### CLOUDSTACK VERSION

~~~
4.18, 4.19 
~~~

##### SUMMARY

When creating a VPC a user adds DNS entries for the VPC, those DNS servers should be used by all network tiers of the VPC, but instead the network tiers still use the Zone

##### STEPS TO REPRODUCE

~~~
Create a new VPC
Enter DNS1 and DNS2 details
Create a network tier
Check DNS entries for the network tier
~~~

##### EXPECTED RESULTS

~~~
Network tiers should use VPC defined DNS entries
~~~

##### ACTUAL RESULTS

~~~
Network tiers use Zone defined DNS entries
~~~

**Comment by alexandremattioli** [comment]
Can you have a look at this please? @weizhouapache

**Comment by weizhouapache** [comment]
> Can you have a look at this please? @weizhouapache

@alexandremattioli sure, I will test it

**Comment by weizhouapache** [comment]
I was able to reproduce the issue

- create vpc with DNS 8.8.8.8 and 1.1.1.1
![image](https://github.com/apache/cloudstack/assets/57355700/830b442a-d522-49f3-ad1e-2f0b15c82d76)

- create vpc tier without DNS, it has DNS servers same the data center

![image](https://github.com/apache/cloudstack/assets/57355700/27f9ed08-715f-4011-9328-7f771acb8152)

- check the VPC VR, it uses DNSs of VPC, but VMs use VR IP and DNS of vpc tier
![image](https://github.com/apache/cloudstack/assets/57355700/9cee4000-67ba-4eb1-9982-a9eaea21ecfb)

**Comment by alexandremattioli** [comment]
Thanks @weizhouapache

**Comment by weizhouapache** [comment]
> Thanks @weizhouapache 

@alexandremattioli 
I will create a fix for it.

**Comment by weizhouapache** [comment]
@alexandremattioli 
PR is created.

**Comment by alexandremattioli** [comment]
@Pearl1594 @nvazquez related to NSX as well

**Comment by DaanHoogland** [comment]
@weizhouapache @alexandremattioli is this solved by #8621 ? (can it be closed)
